src/SonarQube.TeamBuild.PreProcessor/TargetsInstaller.cs

```diff
@@ -114,39 +120,32 @@ private static void CopyIfDifferent(string sourcePath, IEnumerable<string> desti
             }
         }
 
-        private static string GetReadOnlyFileContent(string path)
+        private void WarnOnAppDataInsideSystem32()
         {
-            using (var fs = new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read))
+            // When the scanner is running under Local System or Network Service the AppData folder is under
+            // C:\Windows\System32\... and is ignored by MSBuild and dotnet.
+            if (msBuildPathsSettings.GetImportBeforePaths().Any(IsInsideSystem32))
             {
-                using (var sr = new StreamReader(fs))
-                {
-                    return sr.ReadToEnd();
-                }
+                logger.LogWarning(Resources.MSG_InstallTargetsLocalSystem);
             }
+
+            bool IsInsideSystem32(string path) =>
+                path.IndexOf("windows\\system32", StringComparison.InvariantCultureIgnoreCase) >= 0;
```

Comment:
We have a lot of problems with path comparison (case + separator orientation) so I don't like this comparison. I would recommend to do something like `new FileInfo(path).FullName.IndexOf(Path.Combine("windows", "system32"), StringComparison.InvariantIgnoreCase) >= 0`